core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControlFactory.java

```diff
@@ -13,11 +13,27 @@
  */
 package io.trino.spi.security;
 
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Tracer;
+
 import java.util.Map;
 
 public interface SystemAccessControlFactory
 {
     String getName();
 
+    @Deprecated
     SystemAccessControl create(Map<String, String> config);
+
+    default SystemAccessControl create(Map<String, String> config, SystemAccessControlContext context)
+    {
+        return create(config);
+    }
+
+    interface SystemAccessControlContext
+    {
+        OpenTelemetry getOpenTelemetry();
```

Comment:
Why do you need an instance of OpenTelemetry? Generally, all you need is a properly configured Tracer.

Reply:
I didn't see how access control needs would be different than those of a connector. I followed `ConnectorContext` where @electrum  added both
https://github.com/trinodb/trino/blob/bda3510b195e4177607506f1aa4b4e1ad531c14f/core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorContext.java#L31-L39

Reply:
I couldn't find any authoritative docs or code comments, thus may be drawing wrong conclusions, but both are expected here
https://github.com/airlift/airlift/blob/c55517863fe246f8921b0a03034b0e2f0d7a31c9/http-client/src/main/java/io/airlift/http/client/HttpClientModule.java#L121-L131
and both are later used, if provided.